### DISCUSSION TITLE
post-install hook is not executed if "LastError: Installation error: Compatible mismatch"

### DISCUSSION THREAD

**Discussion by fabianbuettner** [rauc/rauc#911] [General]
Hi,

I have a pre-install hook which is setting up inactive slots as dm-crypt mapped devices for installation, but if "compatible" property does not match, the post-install hook is never running because rauc bundle installation is rejected.
How can I solve this problem?
Is there an error-install hook or something?

best regards,
Fabian

**Comment by jluebbe** [comment]
> I have a pre-install hook which is setting up inactive slots as dm-crypt mapped devices for installation, but if "compatible" property does not match, the post-install hook is never running because rauc bundle installation is rejected. 

Do you mean 'handlers' (supplied by the system) or 'hooks' (supplied by the bundle)?

The post-install hook and slot-post-install handler are documented to only execute on success, so this is as intended.

> How can I solve this problem? Is there an error-install hook or something?

Handling of dm-crypt should be done outside of RAUC:
https://rauc.readthedocs.io/en/latest/faq.html#can-i-use-rauc-with-a-dm-verity-protected-partition

One of the reasons is that (depending on configuration), RAUC will mount slots without a running installation, i.e. to determine the slot state.

An important benefit of that approach is making the dm-crypt key material inaccessible to RAUC.

**Reply by fabianbuettner** [reply]
> Do you mean 'handlers' (supplied by the system) or 'hooks' (supplied by the bundle)?
> 
> The post-install hook and slot-post-install handler are documented to only execute on success, so this is as intended.

I am sorry, yes I am talking about pre install and post install 'handlers'.

Why is the compatible property checked after the re pre install handler was executed? Shouldn't the compatible property be checked before running the pre install handler?


> Handling of dm-crypt should be done outside of RAUC: https://rauc.readthedocs.io/en/latest/faq.html#can-i-use-rauc-with-a-dm-verity-protected-partition
> 
> One of the reasons is that (depending on configuration), RAUC will mount slots without a running installation, i.e. to determine the slot state.
> 
> An important benefit of that approach is making the dm-crypt key material inaccessible to RAUC.

I only need to prepare new partitions, if certain slots are updated, that's why I need to have access to the mounted bundle.
